server/app/controllers/admin/AdminQuestionTranslationsController.java

```diff
@@ -87,28 +82,16 @@ public Result redirectToFirstLocale(Http.Request request, long questionId) {
    *     for the given locale
    */
   @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
-  public CompletionStage<Result> edit(Http.Request request, long questionId, String locale) {
+  public Result edit(Http.Request request, String questionName, String locale) {
     Optional<Locale> maybeLocaleToEdit = translationLocales.fromLanguageTag(locale);
     if (maybeLocaleToEdit.isEmpty()) {
-      return CompletableFuture.completedFuture(
-          redirect(routes.AdminQuestionController.index().url())
-              .flashing("error", String.format("The %s locale is not supported", locale)));
+      return redirect(routes.AdminQuestionController.index().url())
+          .flashing("error", String.format("The %s locale is not supported", locale));
     }
     Locale localeToEdit = maybeLocaleToEdit.get();
 
-    return questionService
-        .getReadOnlyQuestionService()
-        .thenApplyAsync(
-            readOnlyQuestionService -> {
-              try {
-                QuestionDefinition definition =
-                    readOnlyQuestionService.getQuestionDefinition(questionId);
-                return ok(translationView.render(request, localeToEdit, definition));
-              } catch (QuestionNotFoundException e) {
-                return notFound(e.getMessage());
-              }
```

Comment:
(praise) Nice simplification here! And thank you for removing the notFound returning a  simple string!!!

Reply:
😄